content/docs/source-git/how-to-source-git.md

```diff
@@ -5,13 +5,209 @@ draft: false
+  lookaside-cache is also excluded. Upstream sources are part of the working
+  tree in source-git.
+- The `.git` directory of the dist-git repo, and other `.git*` files are also
+  excluded. These are specific to the dist-git repository. If needed, they'll
+  be re-created later on, using a content specific to source-git.
+
+As gitignore rules in the upstream repo and dist-git might be different, I
+need to reset these for the `.distro` directory. In the case of acl for
+example the top-level `.gitignore` has a rule to exclude `Makefile`s, but the
+tests in dist-git have `Makefile` which I want to become part of the
+source-git repo.
+
+Looking at the `.gitignore` file in dist-git, I see that it is only
+source-code archives that are ignored, so there are no other rules to be added
+to `.distro/.gitignore`.
+
+    $ cat > .distro/.gitignore
+    # Reset gitignore rules
+    !
+
+## Configure syncing to dist-git
+
+Now, let's create the configuration which is going to be used when syncing
+content to dist-git. This is used to control certain aspects of syncing
+content to dist-git when using `packit update-dist-git`.
+
+Create a file called `.distro/source-git.yaml` with the following content:
+
+```yaml
+---
+downstream_package_name: acl
+upstream_ref: v2.3.1
+patch_generation_ignore_paths:
+  - .distro/
+sync_changelog: true
+synced_files:
+  - src: .distro/
+    dest: .
+    delete: true
+    filters:
+      - "protect .git*"
+      - "protect sources"
+      - "exclude source-git.yaml"
+      - "exclude .gitignore"
+```
+
+`downstream_package_name` tells `packit` the name of the package.
+
+`upstream_ref` is the Git ref to be used to tell where upstream history ends
+and downstream history starts.
+
+`patch_generation_ignore_paths` lists the paths which should be ignored when
+generating downstream patches, to be added in dist-git. As `.distro` is the
+place where the files related to packaging are stored, I set the
+configuration to ignore this path, so that there are no patch files generated
+for it.
+
+Setting `sync_changelog` to `true` tells Packit to sync the changelog in the
+spec-file as is. This disables Packit's functionality to automatically update
+the changelog when syncing to dist-git.
+
+`synced_files` controls the way files are synced when updating dist-git. In
+the context of the current source-git structure this means to syncing the
+content of `.distro` with the root of the dist-git repo. This is what the
+`src` and `dest` fields configure.
+
+`delete: true` causes files not in `src` to be deleted from `dest`.
+
+`filters` is a list of `rsync` filters to be used to protect paths in `dest`
+and exclude paths in `src` from the sync operation.
+
+You can think about the `synced_files` section as describing the reverse of
+the rsync-command used to copy content from dist-git to `.distro`.
+
+## Update the spec-file and apply the downstream patches
+
+
 # How to create a source-git repository from upstream?
```

Comment:
so we should drop this section now since your guide is more thorough and up to date

I'd only put the link for init somewhere above

Reply:
:+1: